### PR TITLE
Remove 36.0 (LTS) to 51.0 (pre 52.0 LTS) jobdata support

### PR DESCRIPTION
**Pull request by clebergnu**
And include jobdata results and a test for 52.0 (LTS).

The 52.0 release, an LTS release, was meant to allow transitions from
the previous LTS version (36.0lts) and it did so including the ability
to replay jobs created with the older version.

Now, we're 6+ months into the 52.0 release, and as promised, we'll
stop maintaining 36.0lts.  To keep to code clean, let's remove all
this compatibility with versions before 52.0.

Signed-off-by: Cleber Rosa <crosa@redhat.com>

### CODE REVIEW

**Comment by apahim** [comment]
Actually we are 1 month before the 6 months threshold after 52.0 release, but I don't see as a problem removing that compatibility now. Most of them, if not all, are intended for 36to52 migration, so I think we are safe. Reviewing it now.

File: avocado/core/jobdata.py [hunk 1]
```diff
@@ -20,49 +20,22 @@
...
-                             fromlist=[module])
-            return getattr(mod, name)
-        elif module == "avocado.plugins.yaml_to_mux":
-            mod = __import__("avocado_varianter_yaml_to_mux",
-                             fromlist=[module])
-            return getattr(mod, name)
-        else:
-            raise
-
-
```
Comment (apahim) [inline avocado/core/jobdata.py]: ok

File: avocado/core/jobdata.py [hunk 2]
```diff
@@ -150,63 +117,20 @@ def retrieve_variants(resultsdir):
...
-            return get
-        for node in variants.variants.root.iter_children_preorder():
-            node.filters = [[], []]
-            node._environment = None
-            fingerprint = node.fingerprint()
-            node.fingerprint = get_fingerprint_meth("%s%s" % (fingerprint,
-                                                              node.ctrl))
-
-    recorded_mux = _retrieve(resultsdir, VARIANTS_FILENAME + ".json")
-    if recorded_mux:    # new json-based dump
```
Comment (apahim) [inline avocado/core/jobdata.py]: yep, time to go

File: selftests/unit/test_jobdata.py [hunk 3]
```diff
@@ -31,10 +31,10 @@ def _check_results(self, dirname):
...
         act = variants.to_str(0, 99)
         self.assertTrue(act)
-        exp = ("\nVariant variant1-6ec4:    /run/variant1\n"
-               "    /run/variant1:foo => bar\n\n"
-               "Variant variant2-a6fe:    /run/variant2\n"
-               "    /run/variant2:foo => baz")
+        exp = ("\nVariant first-febe:    /run/first\n"
+               "    /run/first:variable_one => 1\n\n"
+               "Variant second-bafe:    /run/second\n"
+               "    /run/second:variable_two => 2")
```
Comment (apahim) [inline selftests/unit/test_jobdata.py]: ok

**Comment by apahim** [comment]
Just to make it clear, we are not dropping the support for v36. It has to be supported until Dec 27, but this cleanup is welcomed now.

**Comment by ldoktor** [comment]
Well I believe this cleanup was suppose to be merged after Dec 27. I mean we can probably go away with it now, but next time please keep things like this after the EOL and not just close to the EOL.

**Comment by clebergnu** [comment]
@ldoktor 52.0 LTS is supposed to have compatibility with 36.0 LTS.  We don't need to keep this baggage on non-LTS versions.  Also, this was on they of other much needed cleanups and improvements.

**Comment by apahim** [comment]
This has nothing to do with 36 EOL. v36 is guaranteed to be supported until Dec 27/2017 and this PR changes nothing in that regard. We will keep backporting fixes until then. And the v52 retro compatibility with v36 is still there. So, what else are you concerned with?

**Pull request merged**